tests/integration/test_charm.py

```diff
@@ -65,3 +65,29 @@ async def test_ingress_reachable(app: Application):
     response = requests.get("http://hello-kubecon")
 
     assert response.status_code == 200
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_owasp_modsecurity_crs(app: Application):
+    """
+    arrange: given charm has been built, deployed, related to a dependent application
+    and owasp-modsecurity-crs is set to True
+    act: when the dependent application is queried via the ingress with malicious request
+    assert: then the response is HTTP 403 Forbidden.
```

Comment:
I think this line should contemplate the 404 case as well, does it make sense?

Reply:
Sure! Done.
```python
assert: then the response is HTTP 403 Forbidden for any request except for /ignore
        whose response is HTTP 404 not found
```